Transports/com.mlapi.contrib.transport.facepunch/Runtime/FacepunchTransport.cs

```diff
@@ -0,0 +1,354 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using MLAPI.Logging;
+using MLAPI.Transports.Tasks;
+using Steamworks;
+using Steamworks.Data;
+using UnityEngine;
+
+namespace MLAPI.Transports.Facepunch
+{
+    using SocketConnection = Steamworks.Data.Connection;
+
+    public class FacepunchTransport : NetworkTransport, IConnectionManager, ISocketManager
+    {
+        private ConnectionManager connectionManager;
+        private SocketManager socketManager;
+        private Dictionary<ulong, Client> connectedClients;
+        private Dictionary<NetworkChannel, SendType> channelSendTypes;
+        private Queue<ConnectionEvent> eventQueue;
+
+        [SerializeField] private List<TransportChannel> channels = new List<TransportChannel>();
+
+        [Space]
+        [Tooltip("The Steam App ID of your game. Technically you're not allowed to use 480, but Valve doesn't do anything about it so it's fine for testing purposes.")]
+        [SerializeField] private uint steamAppId = 480;
+
+        [Tooltip("The Steam ID of the user targeted when joining as a client.")]
+        [SerializeField] public ulong targetSteamId;
+
+        [Header("Info")]
+        [ReadOnly]
+        [Tooltip("When in play mode, this will display your Steam ID.")]
+        [SerializeField] private ulong userSteamId;
+
+        private LogLevel LogLevel => NetworkManager.Singleton.LogLevel;
+
+        private class Client
+        {
+            public SteamId steamId;
+            public SocketConnection connection;
+        }
+
+        private struct ConnectionEvent
+        {
+            public NetworkEvent type;
+            public ulong clientId;
+            public NetworkChannel channel;
+            public ArraySegment<byte> payload;
+            public float receiveTime;
+        }
+
+        #region MonoBehaviour Messages
+
+        private void Awake()
+        {
+            try
+            {
+                SteamClient.Init(steamAppId, false);
+            }
+            catch (Exception e)
+            {
+                if (LogLevel <= LogLevel.Error)
+                    Debug.LogError($"[{nameof(FacepunchTransport)}] - Caught an exeption during initialization of Steam client: {e}");
+            }
+            finally
+            {
+                StartCoroutine(InitSteamworks());
+            }
+        }
+
+        private void FixedUpdate()
```

Comment:
```suggestion
        private void Update()
```

I think this should be in Update. It's not based on physics and calling it in the Update loop makes sense. FixedUpdate could run more frequent or less frequent than the Update loop. This also makes it work much better with a non polling system.